compiler/rustc_next_trait_solver/src/solve/normalizes_to/mod.rs

```diff
@@ -45,7 +45,6 @@ where
                             goal,
                             goal.predicate.alias,
                         );
-                        this.add_goal(GoalSource::AliasWellFormed, goal.with(cx, trait_ref));
```

Comment:
so what's going on is, `proven_via` is `ParamEnv`, but the actual candidate has unsatisfiable region constraints. We then don't add these region constraints when keeping the alias rigid.

Reply:
Correct